pkg/build/controller/strategy/custom.go

```diff
@@ -22,12 +23,22 @@ type CustomBuildStrategy struct {
 
 // CreateBuildPod creates the pod to be used for the Custom build
 func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	data, err := bs.Codec.Encode(build)
+	strategy := build.Spec.Strategy.CustomStrategy
+
+	codec := bs.Codec
+	if strategy.BuildAPIVersion != "" {
+		interfaces, err := latest.InterfacesFor(strategy.BuildAPIVersion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get codec for requested API version (%q): %v", strategy.BuildAPIVersion, err)
```

Comment:
BC validation being risky is a problem solved by having the storage upgrade
path which needs to get turned into a card so we can start sticking things
in it (correcting data in objects to the newest version).  If we drop v1,
we're going to have to touch every object in the system to set it to v2, so
we can update the BC at that time.

On Fri, Jan 22, 2016 at 9:31 AM, Martin Nagy <notifications@github.com>
wrote:

> In pkg/build/controller/strategy/custom.go
> <https://github.com/openshift/origin/pull/6760#discussion_r50542087>:
>
> > @@ -22,12 +23,22 @@ type CustomBuildStrategy struct {
> >
> >  // CreateBuildPod creates the pod to be used for the Custom build
> >  func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
> > -	data, err := bs.Codec.Encode(build)
> > +	strategy := build.Spec.Strategy.CustomStrategy
> > +
> > +	codec := bs.Codec
> > +	if strategy.BuildAPIVersion != "" {
> > +		interfaces, err := latest.InterfacesFor(strategy.BuildAPIVersion)
> > +		if err != nil {
> > +			return nil, fmt.Errorf("failed to get codec for requested API version (%q): %v", strategy.BuildAPIVersion, err)
>
> OK, will update to recognize the error as fatal and add BC validation for
> better user experience.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6760/files#r50542087>.
>